erts/emulator/beam/sys.h

```diff
@@ -1167,14 +1167,14 @@ ERTS_GLB_INLINE size_t sys_strlen(const char *s)
                       (((Uint64)((unsigned char*) (s))[6]) << 8)  | \
                       (((Uint64)((unsigned char*) (s))[7])))
 
-#define put_int64(i, s) do {((char*)(s))[0] = (char)((Sint64)(i) >> 56) & 0xff;\
-                            ((char*)(s))[1] = (char)((Sint64)(i) >> 48) & 0xff;\
-                            ((char*)(s))[2] = (char)((Sint64)(i) >> 40) & 0xff;\
-                            ((char*)(s))[3] = (char)((Sint64)(i) >> 32) & 0xff;\
-                            ((char*)(s))[4] = (char)((Sint64)(i) >> 24) & 0xff;\
-                            ((char*)(s))[5] = (char)((Sint64)(i) >> 16) & 0xff;\
-                            ((char*)(s))[6] = (char)((Sint64)(i) >> 8)  & 0xff;\
-                            ((char*)(s))[7] = (char)((Sint64)(i))       & 0xff;\
+#define put_int64(i, s) do {((char*)(s))[0] = (char)((Sint64)(i) >> 56) & (char)0xff;\
```

Comment:
Nitpick; casting `0xff` to char is implementation-defined.

```suggestion
#define put_int64(i, s) do {((byte*)(s))[0] = (byte)(((Sint64)(i) >> 56) & 0xff);\
```